include/mbedtls/ecp.h

```diff
@@ -762,6 +762,33 @@ int mbedtls_ecp_point_write_binary(const mbedtls_ecp_group *grp,
                                    int format, size_t *olen,
                                    unsigned char *buf, size_t buflen);
 
+/**
+ * \brief           This function exports a point's coordinates.
+ *
+ * \param grp       The group to which the point should belong.
+ *                  This must be initialized and have group parameters
+ *                  set, for example through mbedtls_ecp_group_load().
+ * \param P         The point to export. This must be initialized.
+ * \param X         An initialized MPI. On success, it will contain the
+ *                  X-coordinate of the point in the internal representation
+ *                  used by Mbed TLS.
+ * \param Y         An initialized MPI. On success, it will contain the
+ *                  X-coordinate of the point in the internal representation
+ *                  used by Mbed TLS.
+ * \param Z         An initialized MPI. On success, it will contain the
+ *                  X-coordinate of the point in the internal representation
+ *                  used by Mbed TLS.
+ *
+ * \return          \c 0 on success.
+ * \return          #MBEDTLS_ERR_MPI_ALLOC_FAILED on memory-allocation failure.
+ * \return          Another negative error code on other kinds of failure.
+ */
+int mbedtls_ecp_point_export_coordinates(const mbedtls_ecp_group *grp,
```

Comment:
I was indeed going to argue it's frivolous. I won't fully object to adding an exporter based on this reason - after all, if the exporter is not used, it's going to be garbage-collected by the linker, but my personal preference would be changing the example programs.

I'm slightly inclined to remove "show-off" output from programs, but more importantly, if there's output that's not "show-off" (like, exporting a public key), we should use a standard format for it. This might uncover that we're missing  missing functions for actual use cases.

IMO the risk of adding a generic exporter that also works for "show-off" output is we'll be tempted to use it for cases where an actual application would have to export in a real usable format. (Like, you generated an `ecdsa_keypair` and want to export the public key - you can "show off" that using the coordinates exporter, but in the real world you'd want to export it in a real format that you can `pk_parse()` or `psa_import()` later, and I'm not sure we have a good story for that right now. Edit: well, not this particular one, but this PR is also adding a function for that. But you get the idea.)

Reply:
The ECP interface allows users to build their own cryptographic primitives on top of the core arithmetic. I would expect ECDSA, ECDH and ECJPAKE to only use public ECP functions, since users should be able to write similar things on their own. If not with `mbedtls_ecp_point_export_coordinates`, how could those modules use the result of an operation that calculates a point?

Reply:
Indeed if we consider it a goal for ECP to support people building their own crypto on top, then we need some form of `export_coordinates`.

In my mind that was never really a goal: ECP is for us to build the protocols we implement, and `ecp.h` is only public because that's how we did things back then. But since we never said otherwise (and in particular did not make `ecp.h` internal in 3.0), it's of course perfectly reasonable for people to consider it public, and then the API should be usable not just by us.